pkg/pillar/cmd/zedagent/attesttask.go

```diff
@@ -659,6 +661,7 @@ func attestModuleStart(ctx *zedagentContext) error {
 	log.Functionf("Creating %s at %s", "attestFsmCtx.EnterEventLoop",
 		agentlog.GetMyStack())
 	go ctx.attestCtx.attestFsmCtx.EnterEventLoop()
+	ctx.attestCtx.Started = true
```

Comment:
If I understood correctly, you are trying to solve a race condition here. However, variable assignment is not atomic in Go, so you will need to use a semaphore to completely get rid of the race... otherwise you still cannot guarantee that ctx.attestCtx.Started will be set to true before the `if` checking...

Reply:
I just want to make sure that we started `EnterEventLoop` goroutine and eventually will handle event inside. We run `attestModuleStart` in main routine (as well as `handleEncryptedKeyFromDeviceCreate`), so I do not think that we should be really atomic here.

Reply:
I see, so they cannot be executed in parallel or switched.... understood...

Reply:
LGTM